lib/ios/RNNButtonBuilder.m

```diff
@@ -0,0 +1,55 @@
+#import "RNNButtonBuilder.h"
+#import "RNNFontAttributesCreator.h"
+
+@interface RNNButtonBuilder ()
+@property(weak, nonatomic) UIViewController<RNNLayoutProtocol> *viewController;
+@property(strong, nonatomic) RNNReactComponentRegistry *componentRegistry;
+@end
+
+@implementation RNNButtonBuilder
+
+- (instancetype)initWithComponentRegistry:(id)componentRegistry {
+    self = [super init];
+    self.componentRegistry = componentRegistry;
+    return self;
+}
+
+- (void)bindViewController:(UIViewController<RNNLayoutProtocol> *)viewController {
+    self.viewController = viewController;
+}
+
+- (void)assertButtonId:(RNNButtonOptions *)button {
+    if (!button.identifier.hasValue) {
+        @throw [NSException
+            exceptionWithName:@"NSInvalidArgumentException"
+                       reason:[@"button id is not specified "
+                                  stringByAppendingString:[button.text getWithDefaultValue:@""]]
+                     userInfo:nil];
+    }
+}
+
+- (RNNUIBarButtonItem *)build:(RNNButtonOptions *)button onPress:(RNNButtonPressCallback)onPress {
+    [self assertButtonId:button];
```

Comment:
Typically, a caller function should be placed above the callee. High-level functions are placed above low-level functions.